pandas/core/internals/blocks.py

```diff
@@ -1783,6 +1782,23 @@ def get_values(self, dtype=None):
     def to_dense(self):
         return np.asarray(self.values)
 
+    def to_native_types(self, slicer=None, na_rep="nan", quoting=None, **kwargs):
+        """override to use ExtensionArray astype for the conversion"""
+        values = self.values
+        if slicer is not None:
+            values = values[slicer]
+        mask = isna(values)
+
+        try:
+            values = values.astype(str)
```

Comment:
why would this raise at all? don't we *already* do something like this in the repr of EAs? we should have a same / generic mechanism, rather than implementing logic here.

Reply:
As I explained in the issue (https://github.com/pandas-dev/pandas/issues/28840), it's numpy that can raise an error here. 
But, the reason that this is in a try-except, is not no much for the astype necessarily, but for the setitem on the next line which is not implemented for SparseArray